src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/RaSecondFactorProjector.php

```diff
@@ -262,7 +263,12 @@ public function applySecondFactorVettedEvent(SecondFactorVettedEvent $event)
     {
         $secondFactor = $this->raSecondFactorRepository->find((string) $event->secondFactorId);
 
-        $secondFactor->documentNumber = $event->documentNumber;
+        $documentNumber = null;
```

Comment:
Please move this logic to the concrete implementations of the VettingTypes and create a static `documentNumber` method that way the concrete implementation is responsible for what it will return and you and you can get rid of this logic.